Nef_S2/include/CGAL/Nef_S2/Sphere_circle.h

```diff
@@ -61,15 +63,16 @@ Sphere_circle(const Sphere_point<R>& p, const Sphere_point<R>&q)
 such that a walk along |\Mvar| meets $p$ just before the shorter segment
 between $p$ and $q$. If $p$ and $q$ are antipodal of each other then we
 create any great circle that contains $p$ and $q$.}*/
-{ Point_3 p1(0,0,0), p4 = CGAL::ORIGIN + ((Base*) this)->orthogonal_vector();
+{
   if ( p != q.antipode() ) {
-    if (R_().orientation_3_object()(p1,Point_3(p),
-                                    Point_3(q), p4) != CGAL::POSITIVE )
-      *this = Self(opposite());
+    Point_3 po = CGAL::ORIGIN + Base::orthogonal_vector();
+    //TODO replace Point_3(CGAL::ORIGIN), with CGAL::ORIGIN below
```

Comment:
I think you can just change it as `Point_3` has a constructor from `Origin`.  The same for the other todo further below.

Reply:
It must be explicit, because I had a compile error until I merge your change.